File: lang/python/src/fun/vac/algorithms/number_theory/FibonacciNumber.py
def iterativeProcedure(n: int) -> int:
    sign = 1

    if n < 0:
        if (n & 1) == 0:
            sign = -1
        n = -n

    if n < 2:
        return n

    prev = 0
    curr = 1

    for i in range(2, n + 1):
        prev, curr = curr, prev + curr

    return sign * curr

def recursiveProcedure(n: int) -> int:
    if n < 0:
        if (n & 1) == 0:
            return -recursiveProcedure(-n)
        else:
            return recursiveProcedure(-n)
    elif n < 2:
        return n
    else:
        return recursiveProcedure(n - 2) + recursiveProcedure(n - 1)

File: lang/ruby/src/fun/vac/algorithms/number_theory/FibonacciNumber.rb
module FibonacciNumber

    def self.iterativeProcedure(n)
        sign = 1

        if n.negative?
            if (n & 1).zero?
                sign = -1
            end
            n = -n
        end

        if n < 2
            return n
        end

        prev = 0
        curr = 1

        (2 .. n).each do
            prev, curr = curr, prev + curr
        end

        sign * curr
    end

    def self.recursiveProcedure(n)
        if n.negative?
            if (n & 1).zero?
                return -recursiveProcedure(-n)
            end
            return recursiveProcedure(-n)
        end
        if n < 2
            return n
        end
        recursiveProcedure(n - 2) + recursiveProcedure(n - 1)
    end
end
